tools/shared/src/main/scala/org/scalajs/core/tools/sourcemap/SourceMapWriter.scala

```diff
@@ -100,10 +143,10 @@ class SourceMapWriter(
   }
 
   private def writeHeader(): Unit = {
-    out.write("{\n")
-    out.write("\"version\": 3,\n")
-    out.write("\"file\": " + jsonString(generatedFile) + ",\n")
-    out.write("\"mappings\": \"")
+    val out1 = out
```

Comment:
This caching in local vals looks a bit extreme, even by the standards of the recent changes. Is there evidence that this actually improves performance? I would expect the JVM to optimize that for us.